pkg/secrets/exec_windows.go

```diff
@@ -0,0 +1,108 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018-present Datadog, Inc.
+
+//go:build secrets && windows
+// +build secrets,windows
+
+package secrets
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+
+	"github.com/DataDog/datadog-agent/pkg/util/winutil"
+)
+
+// commandContext sets up an exec.Cmd for running with a context
+func commandContext(ctx context.Context, name string, arg ...string) (*exec.Cmd, func(), error) {
+	cmd := exec.CommandContext(ctx, name, arg...)
+	done := func() {}
+	localSystem, err := getLocalSystemSID()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not query Local System SID: %s", err)
+	}
+	defer windows.FreeSid(localSystem)
+
+	currentUser, err := winutil.GetSidFromUser()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get SID for current user: %s", err)
+	}
+
+	// If we are running as Local System we need to "sandbox" execution to "ddagentuser"
```

Comment:
what if the core agent is running as SYSTEM? This check/drop should not apply to the core agent. Is there a way to differentiate?

Reply:
Thinking about it some more, I believe it is safer to allow continuing in this case. User would have to reconfigure the service to be run with SYSTEM. If we block that we will end up breaking those configs for work with secrets.